test-case/verify-kernel-boot-log.sh

```diff
@@ -23,4 +23,22 @@ disable_kernel_check_point
 
 print_module_params
 
+# Check this device time is NTP Synchronized
+if checkNTPSync; then
+    printf '\nTime Check: NTP Synchronized\n'
+else
+    timedatectl show
+    # try to disable/enable NTP once, this will trigger time sync twice,
+    # before stopping npt and after enabling ntp.
+    # this might fix the problem
+    reenableNTPSync
+
+    if checkNTPSync; then
```

Comment:
Allowing no time at all to synchronize seems very racy and risky. This has to go over the network. I don't know about this new daemon but the old NTP daemon needed at least 1-2 minutes...

Reply:
I think I was right: a short sleep is missing here. Let's say 5 seconds. Here's the logs from NTP failure https://sof-ci.01.org/linuxpr/PR4209/build3362/devicetest/index.html?model=TGLU_RVP_SDW_IPC4ZPH&testcase=verify-kernel-boot-log

As you can see the synchronization failed or timed out the first  time. Rare but it happens. Then this code restarted it (great!) but it did not wait for long enough and missed the success (not great)

```
journalctl --since '2 days ago' -g 'synchro|ntp|timesync'


-- Boot 7ff3328836b8488eadc90078492b2977 --
Feb 22 21:40:43 sh-tglu-rvp-sdw-02 kernel: Mountpoint-cache hash table entries: 32768 (order: 6, 262144 bytes, linear)
Feb 22 21:40:44 sh-tglu-rvp-sdw-02 systemd[1]: Starting Network Time Synchronization...
Feb 22 21:40:44 sh-tglu-rvp-sdw-02 systemd[1]: Started Network Time Synchronization.
Feb 22 21:40:46 sh-tglu-rvp-sdw-02 systemd[1]: tmp-syscheck\x2dmountpoint\x2d023762105.mount: Deactivated successfully.
Feb 22 21:41:06 sh-tglu-rvp-sdw-02 sudo[1814]:   ubuntu : PWD=/home/ubuntu ; USER=root ; COMMAND=/usr/bin/env PATH=/home/ubuntu/sof-test/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin timedatectl set-ntp false
Feb 22 21:41:06 sh-tglu-rvp-sdw-02 systemd-timedated[905]: systemd-timesyncd.service: Disabling unit.
Feb 22 21:41:06 sh-tglu-rvp-sdw-02 systemd[1]: Stopping Network Time Synchronization...
Feb 22 21:41:06 sh-tglu-rvp-sdw-02 systemd-timedated[905]: Set NTP to disabled.
Feb 22 21:41:06 sh-tglu-rvp-sdw-02 systemd[1]: systemd-timesyncd.service: Deactivated successfully.
Feb 22 21:41:06 sh-tglu-rvp-sdw-02 systemd[1]: Stopped Network Time Synchronization.
Feb 22 21:41:06 sh-tglu-rvp-sdw-02 sudo[1840]:   ubuntu : PWD=/home/ubuntu ; USER=root ; COMMAND=/usr/bin/env PATH=/home/ubuntu/sof-test/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin timedatectl set-ntp true
Feb 22 21:41:06 sh-tglu-rvp-sdw-02 systemd-timedated[905]: systemd-timesyncd.service: Enabling unit.
Feb 22 21:41:07 sh-tglu-rvp-sdw-02 systemd-timedated[905]: Set NTP to enabled (systemd-timesyncd.service).
Feb 22 21:41:07 sh-tglu-rvp-sdw-02 systemd[1]: Starting Network Time Synchronization...
Feb 22 21:41:07 sh-tglu-rvp-sdw-02 systemd[1]: Started Network Time Synchronization.
Feb 22 21:41:08 sh-tglu-rvp-sdw-02 systemd-timesyncd[1868]: Initial synchronization to time server 10.104.192.175:123 (corp.intel.com).

```